pkg/informer/informer.go

```diff
@@ -143,34 +193,7 @@ func NewDynamicDiscoverySharedInformerFactory(
 		}
 	}
 
-	crdIsEstablished := func(obj interface{}) bool {
-		crd, ok := obj.(*apiextensionsv1.CustomResourceDefinition)
-		if !ok {
-			utilruntime.HandleError(fmt.Errorf("unable to determine if CRD is established - got type %T instead", obj))
-			return false
-		}
-
-		return apihelpers.IsCRDConditionTrue(crd, apiextensionsv1.Established)
-	}
-
-	// When CRDs change, send a notification that we might need to add/remove informers.
-	crdInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			if crdIsEstablished(obj) {
-				notifyUpdateNeeded()
-			}
-		},
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			oldEstablished := crdIsEstablished(oldObj)
-			newEstablished := crdIsEstablished(newObj)
-			if newEstablished || oldEstablished != newEstablished {
-				notifyUpdateNeeded()
-			}
-		},
-		DeleteFunc: func(obj interface{}) {
-			notifyUpdateNeeded()
-		},
-	})
+	gvrSource.Register(notifyUpdateNeeded)
```

Comment:
yeah, just exposing the updatech is enough here

Reply:
Done